llvm/lib/Target/AMDGPU/SIMemoryLegalizer.cpp

```diff
@@ -2415,6 +2442,16 @@ bool SIGfx12CacheControl::enableVolatileAndOrNonTemporal(
   return Changed;
 }
 
+bool SIGfx12CacheControl::enableLastUse(MachineInstr &MI,
+                                        bool IsLastUse) const {
+  assert(MI.mayLoad() && !MI.mayStore());
+
+  if (IsLastUse && !isScope(MI, AMDGPU::CPol::SCOPE_SYS))
```

Comment:
What's the reason for not doing this for SCOPE_SYS?

Reply:
For simplicity I think we should probably set th=3 regardless of scope.

If I understand correctly "last use" does not really mean anything useful for a system scope load. Normally "last use" means you can discard dirty data from any cache at the same or higher level than the scope. But for system scope, there _is_ no cache at that scope, so there is nothing to do. But it is harmless to set th=3 anyway. @Pierre-vh